src/app/payment/views/IntegratedCheckoutView/CheckoutViewWrapper.tsx

```diff
@@ -182,13 +183,28 @@ const CheckoutViewWrapper = () => {
           return;
         }
         if (checkoutTheme && plan) {
+          if (promotionCode) {
+            handleFetchPromotionCode(plan.selectedPlan.id as string, promotionCode).catch((err) => {
+              const error = err as Error;
+              const errorMessage = error.message.includes('Promotion code with an id')
+                ? error.message
+                : 'Something went wrong, try again later';
```

Comment:
Is there no other way to check this than by checking the erorr message? With a status code or something like that?